src/configuration.js

```diff
@@ -0,0 +1,252 @@
+ * `maxOpenedStorages` | `MAX_OPENED_STORAGES` | `1000`
+ *
+ * ## Not Supported environment variables
+ *
+ * - `MEMORY_MBYTES`
+ * - `HEADLESS`
+ * - `XVFB`
+ * - `CHROME_EXECUTABLE_PATH`
+ */
+export class Configuration {
+    // all env vars also supports `APIFY_` prefix
+    static ENV_MAP = {
+        TOKEN: 'token',
+        LOCAL_STORAGE_DIR: 'localStorageDir',
+        LOCAL_STORAGE_ENABLE_WAL_MODE: 'localStorageEnableWalMode',
+        DEFAULT_DATASET_ID: 'defaultDatasetId',
+        DEFAULT_KEY_VALUE_STORE_ID: 'defaultKeyValueStoreId',
+        DEFAULT_REQUEST_QUEUE_ID: 'defaultRequestQueueId',
+        METAMORPH_AFTER_SLEEP_MILLIS: 'metamorphAfterSleepMillis',
+        PERSIST_STATE_INTERVAL_MILLIS: 'persistStateIntervalMillis',
+        TEST_PERSIST_INTERVAL_MILLIS: 'persistStateIntervalMillis', // for BC, seems to be unused
+        ACTOR_EVENTS_WS_URL: 'actorEventsWsUrl',
+        INPUT_KEY: 'inputKey',
+        ACTOR_ID: 'actorId',
+        API_BASE_URL: 'apiBaseUrl',
+        IS_AT_HOME: 'isAtHome',
+        ACTOR_RUN_ID: 'actorRunId',
+        ACTOR_TASK_ID: 'actorTaskId',
+        CONTAINER_PORT: 'containerPort',
+        CONTAINER_URL: 'containerUrl',
+        USER_ID: 'userId',
+        PROXY_HOSTNAME: 'proxyHostname',
+        PROXY_PASSWORD: 'proxyPassword',
+        PROXY_STATUS_URL: 'proxyStatusUrl',
+        PROXY_PORT: 'proxyPort',
+        MAX_OPENED_STORAGES: 'maxOpenedStorages',
+
+        // not supported, use env vars directly:
+        // MEMORY_MBYTES: 'memoryMbytes',
+        // HEADLESS: 'headless',
+        // XVFB: 'xvfb',
+        // CHROME_EXECUTABLE_PATH: 'chromeExecutablePath',
+    };
+
+    static ENV_MAP_REVERSED = Object.entries(Configuration.ENV_MAP).reduce((obj, [key, value]) => {
+        obj[value] = key;
+        return obj;
+    }, {});
+
+    static BOOLEAN_VARS = ['localStorageEnableWalMode'];
+
+    static INTEGER_VARS = ['proxyPort', 'internalPort', 'memoryMbytes', 'containerPort'];
+
+    static DEFAULTS = {
+        defaultKeyValueStoreId: 'default',
+        defaultDatasetId: 'default',
+        defaultRequestQueueId: 'default',
+        inputKey: 'INPUT',
+        proxyHostname: 'proxy.apify.com',
+        apiBaseUrl: 'https://api.apify.com/v2',
+        proxyStatusUrl: 'http://proxy.apify.com',
+        proxyPort: 8000,
+        containerPort: 4321,
+        containerUrl: 'http://localhost:4321', // Must match `containerPort` above!
+        maxOpenedStorages: 1000,
+        metamorphAfterSleepMillis: 300e3,
+        persistStateIntervalMillis: 60e3, // This value is mentioned in jsdoc in `events.js`, if you update it here, update it there too.
+        localStorageEnableWalMode: true,
+    };
+
+    /**
+     * Creates new `Configuration` instance with provided options. Env vars will have precedence over those.
+     *
+     * @param {Record<string, number | string | boolean>} options
+     */
+    constructor(options = {}) {
+        this.options = options;
```

Comment:
Later in the code, we're adding keys to the object. We should make a copy not to modify the user's object.